backend/src/main/java/eventlistener/security/filter/JwtAuthFilter.java

```diff
@@ -0,0 +1,53 @@
+package eventlistener.security.filter;
+
+import eventlistener.security.service.JWTUtilService;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.List;
+
+@Slf4j
+@Component
+public class JwtAuthFilter extends OncePerRequestFilter {
+
+    private final JWTUtilService jwtUtil;
+
+    public JwtAuthFilter(JWTUtilService jwtUtil) {
+        this.jwtUtil = jwtUtil;
+    }
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain) throws ServletException, IOException {
+        String token = getAuthToken(request);
+        try{
+            if(token != null && !token.isBlank()){
+                String username = jwtUtil.extractUsername(token);
+                setSecurityContext(username);
+            }
+        }catch (Exception e){
+            log.error("No valid Token found!", e);
```

Comment:
i have this line in my quiz app here : 
 response.sendError(HttpServletResponse.SC_FORBIDDEN);
I think there was an issue with  the exception handler not picking up the exception thrown without this line?
Maybe double check with the coaches

Reply:
strange: In my quiz app I have the above posted line and everything works as expected. I spent the entire afternoon now trying to figure out why logging in doesn't work in the capstone app that I set up. And alas it's this very line of code.
Now after removing it it works perfectly. So maybe better don't listen on what I commented earlier